src/Schedulers/SchedulerTraits.php

```diff
@@ -262,8 +262,14 @@ public static function do_action_or_reschedule() {
 		// or schedule to run now if no blocking jobs exist.
 		$blocking_job = static::get_next_blocking_job( $action_name );
 		if ( $blocking_job ) {
+			$schedule = $blocking_job->get_schedule();
+			if ( method_exists( $schedule, 'get_date' ) ) {
```

Comment:
What would happen here so schedule wouldn't have this method? Should we add a comment to clarify?

Reply:
Nice catch. Added in `8646371`.